cxx/irm_test.cc

```diff
@@ -60,3 +57,53 @@ BOOST_AUTO_TEST_CASE(test_irm) {
                           irm.relations.at("R4"));
   BOOST_TEST(r4_obs == 2);
 }
+
+BOOST_AUTO_TEST_CASE(test_irm_one_data_point) {
+  // We have one data point {1, 1}, for which we compute different relations
+  // against. Because each domain (D1, D2) has only one point, there is only one
+  // cluster, and thus no marginalizing over different cluster assignments, so
+  // logp_score = logp(x).
+  std::map<std::string, T_relation> schema1{
+      {"R1", T_clean_relation{{"D1"}, DistributionSpec("bernoulli")}},
+      {"R2", T_clean_relation{{"D2"}, DistributionSpec("normal")}},
+      {"R3", T_clean_relation{{"D1", "D2"}, DistributionSpec("normal")}}};
+  IRM irm(schema1);
+
+  BOOST_TEST(irm.logp_score() == 0.0);
+
+  std::mt19937 prng;
+  irm.transition_cluster_assignments_all(&prng);
+  BOOST_TEST(irm.logp_score() == 0.0);
+
+  auto obs0 = observation_string_to_value("0", ObservationEnum::bool_type);
```

Comment:
these should also work as `bool obs0 = false` etc (I put this in `test_irm` to make sure `observation_string_to_value` was called somewhere, in hindsight it would be better to have a separate test for it, but no need to fix that in this PR)

Reply:
Done.